### PR TITLE
Fix rounding issue with Manipulators\Size

### PR DESCRIPTION
**Pull request by sebastianpopp**
If I have an image with a size of 1024x553 pixel and I run

```php
(new \League\Glide\Manipulators\Size)
    ->setParams(['w' => 411, 'fit' => 'max'])
    ->run($image);
```

on it. I will get an image with the dimensions 409x221 pixel.

It looks like this happens due to the calculations in \League\Glide\Manipulators\Size::resolveMissingDimensions(). For the example above it will keep the width of 411 px and calculate a height of 221 px. Intervention Image rounds numbers differently, resulting in this wrong image size when the image is actually resized to the precalculated size.

To fix this issue I would suggest to rely on the resizing logic of Intervention Image.

### CODE REVIEW

File: tests/Manipulators/SizeTest.php [hunk 1]
```diff
@@ -126,6 +126,16 @@ public function testResolveMissingDimensions()
...
 
+    public function testResolveMissingDimensionsWithOddDimensions()
+    {
+        $image = Mockery::mock('Intervention\Image\Image', function ($mock) {
+            $mock->shouldReceive('width')->andReturn(1024);
+            $mock->shouldReceive('height')->andReturn(553);
+        });
+
+        $this->assertSame([411, 222], $this->manipulator->resolveMissingDimensions($image, 411, null));
+    }
```
Comment (rjd22) [inline tests/Manipulators/SizeTest.php:137]: @sebastianpopp thx the fix. Might it be a good idea to also add tests for a resize with a null width?

**Comment by rjd22** [comment]
I saw the GD issue was fixed so if you rebase it on the last master most likely the CI will pass.

File: tests/Manipulators/SizeTest.php (hunk 1 above)
Comment (sebastianpopp) [reply tests/Manipulators/SizeTest.php:137]: There's already a test for a resize with a null width above. I just wanted to add an additional test case that fails with the old implementation and passes with my fix to demonstrate why it's needed.

**Review by tgalopin — APPROVED** [review]

**Review by ADmad — APPROVED** [review]

**Comment by ADmad** [comment]
Thanks for nudging this along @rjd22 :slightly_smiling_face:.
